### PR TITLE
Fixed compilation in GCC 7

### PR DESCRIPTION
**Pull request by Razican**
As talked in #1300, I added an exception for GCC 7 so that it will not show the false positive uninitialized warning.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -149,6 +149,10 @@ ELSE()
   ADD_COMPILE_FLAG("-Wall")
   ADD_COMPILE_FLAG("-Werror")
   ADD_COMPILE_FLAG("-Wextra")
+  execute_process(COMMAND ${CMAKE_C_COMPILER} -dumpversion OUTPUT_VARIABLE GCC_VERSION)
```
Comment (dschuff) [inline CMakeLists.txt]: Wil this work if `CMAKE_C_COMPILER` is cl.exe (i.e. windows)?

**Comment by kripken** [comment]
As suggested by @binji in https://github.com/WebAssembly/binaryen/issues/1300#issuecomment-345843868 , turns out there are ways to change this warning for a single file in the source code - that seems optimal.

File: CMakeLists.txt (hunk 1 above)
Comment (Razican) [reply CMakeLists.txt]: It should work as this was on the "not Windows" condition branch.

**Comment by Razican** [comment]
I did the suggested solution. It works in my GCC 7.2.

File: src/tools/fuzzing.h [hunk 2]
```diff
@@ -1320,12 +1320,24 @@ class TranslateToFuzzReader {
     return first;
   }
 
+  #define GCC_VERSION (__GNUC__ * 10000 \
```
Comment (kripken) [inline src/tools/fuzzing.h]: Thanks, this looks good. Please just add a comment before this with an explanation and a link to the upstream bug ( https://gcc.gnu.org/bugzilla/show_bug.cgi?id=82800 ).

Also, before we merge, have you joined the [community group](https://www.w3.org/community/webassembly/participants)?

Comment (Razican) [reply src/tools/fuzzing.h]: I added the comment and signed up for the community group :)

**Comment by kripken** [comment]
Great, thanks!

**Pull request merged**